network.go

```diff
@@ -852,6 +852,10 @@ func (n *network) ipamAllocate() ([]func(), error) {
 				if ip = net.ParseIP(v); ip == nil {
 					return nil, types.BadRequestErrorf("non parsable secondary ip address %s (%s) passed for network %s", k, v, n.Name())
 				}
+				if !d.Pool.Contains(ip) {
```

Comment:
I think this should be changed to check if the AuxAddress is outside the PreferredPool and fail.
If it is within the preferredPool then we should be able to allocate it in the larger pool correct ?